--- /tmp/mdread.f	2009-07-17 15:55:53.000000000 -0700
+++ mdread.f	2006-04-03 16:35:47.000000000 -0700
@@ -268,6 +268,14 @@
       gbalpha = ONE
    end if
    
+   !--------------------------------------------------------------------
+   ! If user has requested PB electrostatics, read some more input
+   !--------------------------------------------------------------------
+
+   if ( igb == 10 ) then
+      call pb_read
+   end if
+   
    ! -------------------------------------------------------------------
    ! If the user has requested NMR restraints, do a cursory read of the
    ! restraints file(s) now to determine the amount of memory necessary
@@ -844,6 +852,15 @@
       
    end if  ! ( igb /= 0 .and. igb /= 10)
    
+   !------------------------------------------------------------------------
+   ! If user has requested Poisson-Boltzmann electrostatics, set up variables
+   !------------------------------------------------------------------------
+    
+   if ( igb == 10 ) then
+      call pb_init(ifcap,natom,nres,ntypes,nbonh,nbona,ix(i02),ix(i04),ix(i06),ix(i08),ix(i10),&
+                   ix(iibh),ix(ijbh),ix(iiba),ix(ijba),ih(m02),ih(m04),ih(m06),x(l15),x(l97))
+   end if  ! ( igb == 10 ) 
+ 
    if (icnstph /= 0) then
       !     Read charge data and alter current charges accordingly
       call cnstphread(ix(icpstinf),ix(icpresst),ix(icpptcnt), &
