--- /opt/amber11/src/configure	2009-12-27 06:05:23.000000000 -0800
+++ configure	2010-01-04 14:06:32.000000000 -0800
@@ -192,7 +192,7 @@
         x86_64=yes ;;
     *) ;;
     esac
-else
+elif [ $macAccelerate != "no" ]; then
     echo "This is not a Macintosh, disabling -macAccelerate."
     macAccelerate="no"
 fi
@@ -247,6 +247,13 @@
 ucpp=install
 cpp="\$(BINDIR)/ucpp -l"
 
+#------------------------------------------------------------------------------
+#  Local Overrides
+#------------------------------------------------------------------------------
+if [ -e 'configure_override' ]; then
+   . configure_override
+fi
+
 #-----------------------------------
 # skip building of sleap?
 #-----------------------------------
@@ -657,8 +664,6 @@
 elif [ "$macAccelerate" = "yes" -a "$mkl" = "no" ] ; then
     lapack=skip
     blas=skip
-    flibs="\$(LIBDIR)/carpack.a \$(LIBDIR)/f2c.a"
-    flibsf="\$(LIBDIR)/carpack.a \$(LIBDIR)/f2c.a"
     flibs_arch="-framework Accelerate"
 elif [ "$macAccelerate" = "yes" -a "$mkl" = "yes" ] ; then
     echo "MKL setting presence disables -macAccelerate."
@@ -669,7 +674,7 @@
 #  Support platforms without a C <complex.h> by building c9x-complex.
 #--------------------------------------------------------------------------
 
-if [ -r /usr/include/complex.h ]; then
+if [ -r /usr/include/complex.h -o "$c9xcomplex" = "skip" ]; then
     c9xcomplex='skip'
 else
     c9xcomplex='libmc.a'
